compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -573,7 +573,8 @@ object ProtoTypes {
    */
   private def wildApprox(tp: Type, theMap: WildApproxMap, seen: Set[TypeParamRef])(implicit ctx: Context): Type = tp match {
     case tp: NamedType => // default case, inlined for speed
-      if (tp.symbol.isStatic || (tp.prefix `eq` NoPrefix)) tp
+      if (tp.isInstanceOf[TypeRef] && tp.symbol.is(Flags.Case) && !tp.symbol.isClass) WildcardType(tp.underlying.bounds)
```

Comment:
I don't understand what this condition is testing? Would be good to add a comment or factor out into a well-named val definition.

Reply:
It's testing if `tp` is a pattern-bound type variable - I've extracted the condition into an appropriate variable.